app/lib/actions/pulp3/orchestration/repository/delete.rb

```diff
@@ -9,10 +9,13 @@ def plan(repository, smart_proxy)
               plan_action(Actions::Pulp3::Repository::DeleteDistributions, repository.id, smart_proxy)
 
               if repository.content_view.default?
-                #we're deleting the library instance, so just delete the whole pulp3 repo
+                # Container push repositories must be deleted through the distribution
+                return if repository.root.is_container_push
+
```

Comment:
Why do we need this? Should we be able to delete unpublished container-push repos in pulp?

Reply:
In Pulp Container, a container push repo is tightly-coupled to its distribution. A container push repository cannot be deleted in fact, only its distribution can be deleted.